src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResultSimulationResource.java

```diff
@@ -0,0 +1,95 @@
+            ProgrammingSubmissionResultSimulationService programmingSubmissionResultSimulationService) {
+        this.programmingSubmissionService = programmingSubmissionService;
+        this.userService = userService;
+        this.participationService = participationService;
+        this.messagingService = messagingService;
+        this.programmingExerciseService = programmingExerciseService;
+        this.programmingSubmissionResultSimulationService = programmingSubmissionResultSimulationService;
+    }
+
+    /**
+     * This method is used to create a participation and a submission
+     * This participation and submission are only SIMULATIONS for the testing
+     * of programming exercises without local setup
+     *
+     * @param exerciseID the id of the exercise
+     * @return HTTP OK and ProgrammingSubmission
+     */
+
+    @PostMapping(value = "submissions/no-local-setup/{exerciseID}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ProgrammingSubmission> createParticipationAndSubmissionSimulation(@PathVariable Long exerciseID) {
+
+        ProgrammingSubmission programmingSubmission = programmingSubmissionResultSimulationService.createSubmission(exerciseID);
+
+        programmingSubmissionService.notifyUserAboutSubmission(programmingSubmission);
+
+        return ResponseEntity.ok().body(programmingSubmission);
+    }
+
+    /**
+     * This method is used to notify artemis that there is a new programming exercise build result.
+     * This result is only a SIMULATION for the testing of programming exercises without local setup
+     *
+     * @param exerciseID id of the exercise
+     * @return HTTP OK and Result
+     */
+    @PostMapping(value = "results/no-local-setup/{exerciseID}")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Result> createNewProgrammingExerciseResult(@PathVariable Long exerciseID) {
+        log.debug("Received result notify (NEW)");
+        User user = userService.getUserWithGroupsAndAuthorities();
+        Participant participant = user;
+        ProgrammingExercise programmingExercise = programmingExerciseService.findByIdWithEagerStudentParticipationsAndSubmissions(exerciseID);
+        Optional<StudentParticipation> optionalStudentParticipation = participationService.findOneByExerciseAndParticipantAnyState(programmingExercise, participant);
+        ProgrammingExerciseStudentParticipation programmingExerciseStudentParticipation = (ProgrammingExerciseStudentParticipation) optionalStudentParticipation.get();
+        Result result = programmingSubmissionResultSimulationService.createResult(programmingExerciseStudentParticipation);
+
+        messagingService.broadcastNewResult((Participation) optionalStudentParticipation.get(), result);
+        // TODO: can we avoid to invoke this code for non LTI students? (to improve performance) // if (participation.isLti()) { // }
```

Comment:
please remove the comment, this will never work for LTI

Reply:
i removed the comment